doc/modules/classification_threshold.rst

```diff
@@ -0,0 +1,171 @@
+    >>> classifier = DecisionTreeClassifier(max_depth=2, random_state=0).fit(X, y)
+    >>> classifier.predict_proba(X[:4])
+    array([[0.94   , 0.06   ],
+           [0.94   , 0.06   ],
+           [0.04..., 0.95...],
+           [0.04..., 0.95...]])
+    >>> classifier.predict(X[:4])
+    array([0, 0, 1, 1])
+
+While these approaches are reasonable as default behaviors, they are not be ideal for
+all cases. The context and nature of the use case defines the expected behavior of the
+classifier and thus the strategy to convert soft predictions into hard predictions. We
+illustrate this point with an example.
+
+Let's imagine the deployment of a predictive model helping medical doctors to detect
+tumour. In a setting where this model was a tool to discard obvious cases and false
+positives don't lead to potentially harmful treatments, doctors might be interested in
+having a high recall (all cancer cases should be tagged as such) to not miss any patient
+with a cancer. However, that is at the cost of having more false positive predictions
+(i.e. lower precision). Thus, in terms of decision threshold, it may be better to
+classify a patient as having a cancer for a posterior probability estimate lower than
+0.5.
+
+Post-tuning of the decision threshold
+=====================================
+
+One solution to address the problem stated in the introduction is to tune the decision
+threshold of the classifier once the model has been trained. The
+:class:`~sklearn.model_selection.TunedThresholdClassifier` tunes this threshold using an
+internal cross-validation. The optimum threshold is chosen to maximize a given metric
+with or without constraints.
+
+The following image illustrates the tuning of the cut-off point for a gradient boosting
+classifier. While the vanilla and tuned classifiers provide the same Receiver Operating
+Characteristic (ROC) and Precision-Recall curves, and thus the same
+:term:`predict_proba` outputs, the class label predictions differ because of the tuned
+decision threshold. The vanilla classifier predicts the class of interest for a
+posterior probability greater than 0.5 while the tuned classifier predicts the class of
+interest for a very low probability (around 0.02). This cut-off point optimizes a
+utility metric defined by the business case (in this case an insurance company).
+
+.. figure:: ../auto_examples/model_selection/images/sphx_glr_plot_tuned_threshold_classifier_002.png
+   :target: ../auto_examples/model_selection/plot_tuned_threshold_classifier.html
+   :align: center
+
+Available options to tune the cut-off point
+-------------------------------------------
+
+The cut-off point can be tuned with different strategies controlled by the parameter
+`objective_metric`.
+
+A straightforward use case is to maximize a pre-defined scikit-learn metric. These
+metrics can be found by calling the function :func:`~sklearn.metrics.get_scorer_names`.
+We provide an example where we maximize the balanced accuracy.
```

Comment:
```suggestion
In this example, we maximize the balanced accuracy.
```

Reply:
Does this sentence refer to the below example? because the example optimizes the f1 score.